plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/move/MoveTypeRefactoringParticipant.java

```diff
@@ -48,19 +48,20 @@ public class MoveTypeRefactoringParticipant extends MoveParticipant {
 	private String oldPackageName;
 	private String newPackageName;
 	private TypeEntry type;
-	private URI destinationURI;
-	private IFile file;
+	private IFile destinationFile;
+	private IFile currentFile;
 
 	@Override
 	protected boolean initialize(final Object element) {
 		if (element instanceof final IFile file) {
-			this.file = file;
+			this.currentFile = file;
 			this.type = TypeLibraryManager.INSTANCE.getTypeEntryForFile(file);
 			this.oldPackageName = type.getPackageName();
 			if (getArguments().getDestination() instanceof final IResource folder) {
-				this.destinationURI = URI.createPlatformResourceURI(folder.getFullPath().toString(), true)
+				final URI destinationURI = URI.createPlatformResourceURI(folder.getFullPath().toString(), true)
 						.appendSegment(type.getURI().lastSegment());
-				this.newPackageName = PackageNameHelper.getPackageNameFromFile(getFileFromURI(destinationURI));
+				this.destinationFile = getFileFromURI(destinationURI);
```

Comment:
I suggest avoiding the detour via the URI entirely:
```java
			if (getArguments().getDestination() instanceof final IContainer destination) {
				this.destinationFile = destination.getFile(IPath.fromOSString(file.getFullPath().lastSegment()));
```

Reply:
thx for pointing that out  :) .. actually i did not know how to do that so this my workaround

Reply:
Please add this to the follow up PR https://github.com/eclipse-4diac/4diac-ide/pull/655/files